Yoast/Sniffs/WP/RequiredParametersSniff.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * YoastCS\Yoast\Sniffs\Files\FileNameSniff.
+ *
+ * @package Yoast\YoastCS
+ * @author  Jip Moors
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace YoastCS\Yoast\Sniffs\WP;
+
+use WordPress\AbstractFunctionParameterSniff;
+
+/**
+ * Verifies that optional parameters are set, when they should be set.
+ *
+ * @package Yoast\YoastCS
+ * @author  Jip Moors
+ *
+ * @since   0.6
+ */
+class RequiredParametersSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'wp_required_parameters';
+
+	/**
+	 * Functions this sniff is looking for.
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *    $target_functions = array(
+	 *        (string) Function name. => array(
+	 *            (int) Target parameter position, 1-based. => array(
+	 *                'name'    => (string)  The name of the argument that should be set.
+	 *                'enforce' => (boolean) If this should trigger an error or warning.
+	 *            )
+	 *         )
+	 *    );
+	 */
+	protected $target_functions = array(
+		'update_option' => array(
+			3 => array(
+				'name'    => 'autoload',
+				'enforce' => false,
+			),
+		),
+		'add_option'    => array(
+			4 => array(
+				'name'    => 'autoload',
+				'enforce' => true,
+			),
+		),
+	); // End $target_functions.
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * This method has to be made concrete in child classes.
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return int|void Integer stack pointer to skip forward or void to continue
+	 *                  normal file processing.
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		foreach ( $this->target_functions[ $matched_content ] as $position => $parameter_args ) {
+			// Check that number of parameters defined is not less than the position to check.
+			if ( isset( $parameters[ $position ] ) ) {
+				continue;
+			}
+
+			$message = '%s() is expected to be called with the "%s" argument at the %d position.';
```

Comment:
`at the 3 position` is still grammatically incorrect. `at position %d` is grammatically correct.